src/main/java/com/exadel/discount/service/impl/DiscountServiceImpl.java

```diff
@@ -65,20 +66,43 @@ public DiscountDTO getById(UUID id) {
     @Override
     public List<DiscountDTO> getAll() {
         log.debug("Getting list of all Discounts");
-        List<DiscountDTO> discountDTOS = discountMapper.getListDTO(discountRepository.findAll());
+        List<DiscountDTO> discountDTOS = discountMapper.getListDTO(discountRepository.findAllByArchivedFalse());
         log.debug("Successfully got list of all Discounts");
         return discountDTOS;
     }
 
     @Override
+    @Transactional
     public void deleteById(UUID id) {
         log.debug(String.format("Deleting Discount with ID %s", id));
-        discountRepository.findById(id).orElseThrow(() ->
+        Discount discount = discountRepository.findByIdAndArchivedFalse(id).orElseThrow(() ->
```

Comment:
it will be good to keep the same style for each time you use Optional.
For some methods, you splitting line on `.orElse`, for some not.